src/pages/home/HeaderView.tsx

```diff
@@ -143,9 +143,11 @@ function HeaderView({report, personalDetails, parentReport, parentReportAction,
         Report.updateNotificationPreference(reportID, report.notificationPreference, CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS, false, report.parentReportID, report.parentReportActionID),
     );
 
-    const canJoinOrLeave = !isSelfDM && !isGroupChat && (isChatThread || isUserCreatedPolicyRoom || canLeaveRoom || canLeavePolicyExpenseChat);
-    const canJoin = canJoinOrLeave && !isWhisperAction && report.notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
-    const canLeave = canJoinOrLeave && ((isChatThread && !!report.notificationPreference?.length) || isUserCreatedPolicyRoom || canLeaveRoom || canLeavePolicyExpenseChat);
+    const canLeave =
```

Comment:
@nkdengineer Any ideas on tidying up `canLeave`? It would be nice if we could get rid of this duplicated set of conditions `isUserCreatedPolicyRoom || canLeaveRoom || canLeavePolicyExpenseChat`.

Reply:
I would suggest we create ReportUtils helper methods and just avoid using too many ternaries and stuff.

There's chats you can leave and chats you can join. Let's stop combining the logic for both of them.

Reply:
Let's do `canLeaveChat()` and `canJoinChat()`.

Reply:
@jjcoffee @marcaaron I updated to create `canLeaveChat` and `canJoinChat` util.